include/zephyr/devicetree/mem.h

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ * @brief Memory Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2020, Linaro Ltd.
+ * Copyright (c) 2020 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_MEM_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_MEM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/devicetree/memory-attr.h>
+
+/**
+ * @defgroup devicetree-memory Devicetree memory API
+ * @ingroup devicetree
+ * @{
+ */
+
+/* Inidicates that a ram region is nocache. */
+/* TODO Obtained from zephyr,memory-region.yaml. This should be defined in
+ * just one place.
+ */
+#define MEM_NOCACHE_ATTR "RAM_NOCACHE"
+
+/**
+ * @brief Container for memory-region information specified in devicetree
+ *
+ * This type contains pointers to the begin and end of the memory region and
+ * a string that specifies such region's properties.
+ *
+ */
+struct dt_mem_region {
+	const uint8_t *start;
+	const uint8_t *end;
+	const char *region_mpu;
```

Comment:
maybe call it `region_attr`

Reply:
I have replaced it by a `bool` as per your comment below.